Mathlib/GroupTheory/PushoutI.lean

```diff
@@ -183,3 +190,486 @@ variable [∀ i, Group (G i)] [Group H] {φ : ∀ i, H →* G i}
 instance : Group (PushoutI φ) :=
   { Con.group (PushoutI.con φ) with
     toMonoid := PushoutI.monoid }
+
+namespace NormalWord
+
```

Comment:
I think it would be good to add a good chunk of documentation here, explaining what is true, what definitions you are going to introduce, and how you will prove that the normal form is indeed a normal form.